pintc/src/error/compile_error.rs

```diff
@@ -1229,6 +1236,23 @@ impl ReportableError for CompileError {
                 color: Color::Red,
             }],
 
+            IdenticalPredicates {
+                original_span,
+                span,
+                ..
+            } => vec![
+                ErrorLabel {
+                    message: "original declaration here".to_string(),
+                    span: original_span.clone(),
+                    color: Color::Red,
+                },
+                ErrorLabel {
+                    message: "identical constraints here".to_string(),
```

Comment:
```suggestion
                    message: "identical predicate here".to_string(),
```

Reply:
I worry that this may not be clear to the user. 

Ex.
```
predicate Foo() {
}
predicate Bar() {
    let y = 2;
}
```
will still pop the error.

Reply:
Yeah good point. The trouble is that, while constraints may seem identical, some `let`s may be different and that affects the addresses. So we need a more generalized way of indicating what's going on.

maybe:
"predicate with identical bytecode here"

and then in the note:

"two predicates in a contract cannot have the exact same (optimized) bytecode"